vault-plugins/shared/backent-utils/path_utils.go

```diff
@@ -37,3 +39,21 @@ func ResponseErrMessage(req *logical.Request, message string, status int) (*logi
 	rr := logical.ErrorResponse(message)
 	return logical.RespondWithStatusCode(rr, req, status)
 }
+
+func GetCreationID(expectID bool, data *framework.FieldData) (string, error) {
+	var id string
+
+	if expectID {
+		// for privileged access
+		id = data.Get("uuid").(string)
+		if id == "" {
+			return "", fmt.Errorf("uuid is requered")
+		}
+	}
+
+	if id == "" {
+		id = uuid.New()
+	}
```

Comment:
I think we should not give up on first error from `uuid.New()`. I suggest making one more attempt.